app/src/main/java/com/github/polypoly/app/base/game/service/TaxService.kt

```diff
@@ -0,0 +1,37 @@
+package com.github.polypoly.app.base.game.service
+
+import android.location.Location
+import com.github.polypoly.app.data.GameRepository
+import com.github.polypoly.app.utils.Constants
+
+/**
+ * A service that runs in the background, receives location updates and taxes the [Player]
+ * if they are in a [LocationProperty] that is owned by another [Player].
+ */
+class TaxService: LocationService() {
+    override fun processLocationUpdate(location: Location) {
+        val playerLocation = location // For clarity
+
+        // Iterate through all the locations and check if the player is close enough to interact
+        for (inGameLocation in GameRepository.game?.inGameLocations!!) {
+            // Create a location object from the in-game location
+            val propertyLocation = Location("").apply {
```

Comment:
I think that instead of creating a location that is only used to calculate the distance, you could add in `LocationProperty` a function `distanceTo(location: Location)`  that do the same things, because I think we need to compute this distance many times in the code!